tests/test_metrics/test_keypoint_nme.py

```diff
@@ -0,0 +1,197 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import numpy as np
+from unittest import TestCase
+
+from mmeval.metrics import KeypointNME
+
+
+class TestKeypointNME(TestCase):
+    coco_sigmas = np.array([
+        0.026, 0.025, 0.025, 0.035, 0.035, 0.079, 0.079, 0.072, 0.072, 0.062,
+        0.062, 0.107, 0.107, 0.087, 0.087, 0.089, 0.089
+    ]).astype(np.float32)
+    coco_keypoint_id2name = {
+        0: 'nose',
+        1: 'left_eye',
+        2: 'right_eye',
+        3: 'left_ear',
+        4: 'right_ear',
+        5: 'left_shoulder',
+        6: 'right_shoulder',
+        7: 'left_elbow',
+        8: 'right_elbow',
+        9: 'left_wrist',
+        10: 'right_wrist',
+        11: 'left_hip',
+        12: 'right_hip',
+        13: 'left_knee',
+        14: 'right_knee',
+        15: 'left_ankle',
+        16: 'right_ankle',
+    }
+    coco_dataset_meta = {
+        'dataset_name': 'coco',
+        'num_keypoints': 17,
+        'sigmas': coco_sigmas,
+        'keypoint_id2name': coco_keypoint_id2name
+    }
+
+    def _generate_data(self,
+                       batch_size: int = 1,
+                       num_keypoints: int = 5,
+                       norm_item: str = 'box_size'):
+        """Generate data_batch and data_samples according to different
+        settings."""
+        self.predictions = []
+        self.groundtruths = []
+
+        for i in range(batch_size):
+            keypoints = np.zeros((1, num_keypoints, 2))
+            keypoints[0, i] = [0.5 * i, 0.5 * i]
+            keypoints_visible = np.ones((1, num_keypoints)).astype(bool)
+            keypoints_visible[0, (2 * i) % 8] = False
+
+            prediction = {'coords': keypoints}
+            groundtruth = {
+                'coords': keypoints,
+                'mask': keypoints_visible,
+                norm_item: np.random.random((1, 1)) * 20 * i
+            }
+
+            self.predictions.append(prediction)
+            self.groundtruths.append(groundtruth)
+
+    def test_nme_evaluate(self):
+        """test NME evaluation metric."""
+        # test when norm_mode = 'use_norm_item'
+        # test norm_item = 'box_size' like in `AFLWDataset`
+        norm_item = 'box_size'
+        self.aflw_dataset_meta = {
+            'dataset_name': 'aflw',
+            'num_keypoints': 19,
+            'sigmas': np.array([]),
+        }
+        self._generate_data(
+            batch_size=4, num_keypoints=19, norm_item=norm_item)
+
+        nme_metric = KeypointNME(
+            norm_mode='use_norm_item',
+            norm_item=norm_item,
+            dataset_meta=self.aflw_dataset_meta,
+        )
+        nme_results = nme_metric(self.predictions, self.groundtruths)
+        target = {'NME': 0.0}
+        self.assertDictEqual(nme_results, target)
```

Comment:
There are 3 issues with these test cases:

1. There should be a test case with multiple (>=2) `add` and then `compute`. You can split the existing data into 2 pieces to easily achieve this.
2. `assertDictEqual` is too strong, because there might be floating point precision issues. It's recommended to define a tolerance limit, as done in `np.testing.assert_allclose`.
3. Better to make `predictions` different from `groundtruths`.  Adding a constant offset should work.